framework/base/src/abi/contract_abi.rs

```diff
@@ -16,6 +19,21 @@ pub struct ContractAbi {
 }
 
 impl ContractAbi {
+    pub fn new(build_info: BuildInfoAbi, docs: &[&str], name: &str, has_callback: bool) -> Self {
+        ContractAbi {
+            build_info,
+            docs: docs.iter().map(|s| s.to_string()).collect(),
+            name: name.to_string(),
+            constructors: Vec::new(),
```

Comment:
..Default::default() ? to make it shorter

Reply:
Sure, but I'll leave it for a later PR. I also think a builder patter might make it look nicer, but this requires careful synchronization with the code generator, it's not a prioriry right now.